### ISSUE TITLE
Experiment invoking reconcilers Reconcile method directly in tests instead of registering them as reconcilers

### ISSUE THREAD

**Issue by danail-branekov** [cloudfoundry/korifi#2874]
All our reconcilers are actually delegates of the patching reconcilers, so they are not real reconcilers anyway.

We could test the patching reconciler by registering is the usual way using a dummy delegate to validate its behaviours
For all other reconcilers we could prepare the test setup in BeforeEach blocks and then explicitly call ReconcileResource. This will give us full controll over the setup and will eliminate:
- complex test suites that duplicate the wiring in main.go (this is tested by the e2es)
- simulation of other reconcilers
- the need to use Eventually and Consistently all over the place, since explicit invocation is deterministic

Probably the same can be done for webhooks, but since they are not invoked in a loop and they are not async it is questionable whether it is worth the effort

**Comment by danail-branekov** [comment]
We experimented applying this approach to a the controllers/controllers/services suite as it is not too complex. Here are our thoughts on it:

Pros:
- The suite setup becomes very simple as we stop wiring controllers there
- Tests become more readable and idiomatic to ginkgo and BDD since all prerequsites to the reconcile method are defined in the before each block. In the current setup we have to take other reconcileres into account which makes test definitions more obscure.
- As we are invoking the reconcile method in the `JustBeforeEach` all `Eventually` and `Consistently` statements can go, as the tests become synchronous.

Cons:
- As we stop registering reconcilers with the manager we lose the `Watches` behaviour that makes sure that change to a resource dependency results in a transitive invocation of the reconcile method. This is a big drawback of this approach since it is often the case that we discover those transitive dependencies as we are test driving our change in the controller. If we lose this realistic behaviour we might miss such a transitive dependency between resources. This could be covered by the e2e tests but it would be too expensive.

In summary we think we need to keep the controllers tests in their current form for the time being.
